File: src/main/java/com/examportal/dto/SubmitExamDTO.java
package com.examportal.dto;

import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SubmitExamDTO {
    
    @NotNull(message = "Exam ID must be provided")
    private Long examId;
    
    @NotEmpty(message = "Answers must be provided")
    @Valid
    private List<AnswerDTO> answers;
}

File: Dotnet3/ExamPortal/DTOs/SubmitExamDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class SubmitExamDTO
    {
        [Required(ErrorMessage = "Exam ID must be provided")]
        public long ExamId { get; set; }
        
        [Required(ErrorMessage = "Answers must be provided")]
        public List<AnswerDTO> Answers { get; set; } = new List<AnswerDTO>();
    }
}
